src/data_access/api/SpotifyAPI.java

```diff
@@ -0,0 +1,109 @@
+package data_access.api;
+
+import entity.CommonSongFactory;
+import entity.Song;
+import entity.SongFactory;
+import io.github.cdimascio.dotenv.Dotenv;
+import okhttp3.*;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+public class SpotifyAPI implements SongAPI{
+    private final String CLIENT_ID;
+    private final String CLIENT_SECRET;
+    private final OkHttpClient client;
+    private String authToken = null;
+    private final SongFactory songFactory;
+
+    public SpotifyAPI(SongFactory songFactory){
+        Dotenv dotenv = Dotenv.configure().filename("api.env").load();
+        this.CLIENT_ID = dotenv.get("CLIENT_ID");
+        this.CLIENT_SECRET = dotenv.get("CLIENT_SECRET");
+        this.client = new OkHttpClient().newBuilder().build();
+        this.songFactory = songFactory;
+    }
+
+    @Override
+    public Song getRandomSongFromGenre(String genre) {
```

Comment:
Just a small note about common design adherence with that spacing of the bracket after a function header. I believe a more consistent spacing will improve our general design.